enterprise/users/src/main/java/io/crate/operation/auth/HostBasedAuthentication.java

```diff
@@ -71,13 +72,17 @@
      */
     private Map<String, Map<String, String>> hbaConf;
 
-    private final Map<String, Supplier<AuthenticationMethod>> authMethodRegistry = new HashMap<>();
+    private final Map<String, Function<Protocol, AuthenticationMethod>> authMethodRegistry = new HashMap<>();
 
     @Inject
     public HostBasedAuthentication(Settings settings, UserLookup userLookup) {
         hbaConf = convertHbaSettingsToHbaConf(AuthSettings.AUTH_HOST_BASED_CONFIG_SETTING.setting().get(settings));
-        authMethodRegistry.put(TrustAuthenticationMethod.NAME, () -> new TrustAuthenticationMethod(userLookup));
-        authMethodRegistry.put(ClientCertAuth.NAME, () -> new ClientCertAuth(userLookup));
+        authMethodRegistry.put(TrustAuthenticationMethod.NAME,
+            (protocol) -> new TrustAuthenticationMethod(userLookup));
+        authMethodRegistry.put(ClientCertAuth.NAME,
+            (protocol) -> new ClientCertAuth(userLookup));
+        authMethodRegistry.put(PasswordAuthenticationMethod.NAME,
+            (protocol) -> Protocol.POSTGRES.equals(protocol) ? new PasswordAuthenticationMethod(userLookup) : null);
```

Comment:
IMHO it would be much clearer if you had a lookup method in `HostBasedAuthentication` which is called from resolveAuthenticationType. Like so: 

```java
switch(name) {
  case (TrustAuthenticationMethod.NAME):
     ...
  case (PasswordAuthenticationMethod.NAME):
     if(protocol) {
         ...
     }
}
```

The method registry really should only hold the authentication types but not do the logic of when which method is used.